TESTS/netsocket/socket_sigio/main.cpp

```diff
@@ -65,16 +65,20 @@ bool find_substring(const char *first, const char *last, const char *s_first, co
 void get_data(TCPSocket* sock){
     bool result = false;
     // Server will respond with HTTP GET's success code
-    const int ret = sock->recv(buffer, sizeof(buffer) - 1);
-    if(ret <= 0)
-        return;
-
-    buffer[ret] = '\0';
+    int len = 0;
+    int ret;
+    while((ret = sock->recv(buffer+len, sizeof(buffer) - 1 - len)) > 0) {
+      len += ret;
+    }
+    if(len >= 0)
```

Comment:
Don't need this if - `len` can't be negative if you're only adding positive numbers to it. (And if it could be negative here, I think find_substring would explode below anyway).